std/math/ff.asm

```diff
@@ -0,0 +1,43 @@
+/// Inverts `x` in the finite field with modulus `modulus`.
+/// Assumes that `modulus` is prime, but does not check it.
+let inverse = |x, modulus|
+    if x <= 0 || x >= modulus {
+        std::check::panic("Tried to compute the inverse of zero, of a negative number or a number outside the field.")
+    } else {
+        reduce(extended_gcd(x, modulus)[0], modulus)
+    };
+
+/// Computes `x + y` modulo the modulus.
+let add = |x, y, modulus| reduce(x + y, modulus);
+
+/// Computes `x - y` modulo the modulus.
+let sub = |x, y, modulus| reduce(x - y, modulus);
+
+/// Computes `x * y` modulo the modulus.
+let mul = |x, y, modulus| reduce(x * y, modulus);
+
+/// Computes `x / y` modulo the modulus.
+let div = |x, y, modulus| mul(x, inverse(y, modulus), modulus);
+
+/// Reduces `x` modulo `modulus`, so that it is in the range
+/// between `0` and `modulus`. Works on negative `x`.
+let reduce = |x, modulus|
+    if x < 0 {
+        (modulus - ((-x) % modulus)) % modulus
```

Comment:
We could get rid of the outer modulus by checking if `(modulus - ((-x) % modulus))` is equal to `modulus` and returning 0 in that case. Think this would be more complicated to write but might be faster. Not sure if it's worth it.

Reply:
I tried to add
```
else if x > -modulus { x + modulus}
```
but it was slower

Reply:
I mean it made `inverse` slower. I think during gcd computation, the number should not be much smaller than -modulus, should it?

Reply:
Hm, maybe it doesn't even happen at all...